[BUG] Java cudf build fails in non-conda environment with unable to find rapids-cmake

**Describe the bug**
Building the cudf Java jar in the non-conda Docker environment fails with the following error:
```
-- CPM: adding package cuco@0 (0d602ae21ea4f38d23ed816aa948453d97b2ee4e)
CMake Error at build/_deps/cuco-src/CMakeLists.txt:25 (include):
  include could not find requested file:

    rapids-cmake


CMake Error at build/_deps/cuco-src/CMakeLists.txt:26 (include):
  include could not find requested file:

    rapids-cpm


CMake Error at build/_deps/cuco-src/CMakeLists.txt:27 (include):
  include could not find requested file:

    rapids-cuda


CMake Error at build/_deps/cuco-src/CMakeLists.txt:28 (include):
  include could not find requested file:

    rapids-export


CMake Error at build/_deps/cuco-src/CMakeLists.txt:29 (include):
  include could not find requested file:

    rapids-find


CMake Error at build/_deps/cuco-src/CMakeLists.txt:33 (rapids_cuda_init_architectures):
  Unknown CMake command "rapids_cuda_init_architectures".
```

This appears to have been triggered by #8476.

**Steps/Code to reproduce bug**
Follow the steps in `java/ci/README.md` to build outside of the conda environment in the Docker container

**Expected behavior**
The Java non-conda environment build succeeds.

--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -28,6 +28,17 @@ elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
   set(CUDF_BUILD_FOR_DETECTED_ARCHS TRUE)
 endif()
 
+file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-21.10/RAPIDS.cmake
+    ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-cuda)
+include(rapids-export)
+include(rapids-find)
+
+
 project(CUDF VERSION 21.10.00 LANGUAGES C CXX)
 
 # Needed because GoogleBenchmark changes the state of FindThreads.cmake,

